logrusr/logger.go

```diff
@@ -0,0 +1,143 @@
+package logrusr
+
+import (
+	"fmt"
+	"github.com/activeshadow/logr/util"
+	"github.com/go-logr/logr"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+var (
+	verbosity int
+	loggers   []string
+)
+
+func SetVerbosity(v int) {
+	verbosity = v
+}
+
+func LimitToLoggers(names ...string) {
+	loggers = append(loggers, names...)
+}
+
+type clock struct {
+	mock time.Time // set in tests
+}
+
+func (this *clock) now() time.Time {
+	if this == nil { // normal operations
+		return time.Now()
+	}
+
+	return this.mock // testing only
+}
+
+type LogrusInfoLogr struct {
+	enabled bool
+	name    string
+	kvs     map[string]interface{}
+	logger  logrus.Logger
+
+	// Normally nil, set by test code only.
+	clock *clock
+}
+
+func (this LogrusInfoLogr) Info(msg string, kvs ...interface{}) {
+	if !this.enabled {
+		return
+	}
+
+	logger := this.logger.WithFields(logrus.Fields{
+		"request": &logrus.Fields{
+			"name": this.name,
+			"kvs":  kvs,
+			"this_kvs": this.kvs,
```

Comment:
Would it make sense to merge key/value pairs? You combine them in the `WithValues` function below, so why not here too?

Reply:
I think you're right here. From memory, I think `this.kvs` is added when you create the logger and `kvs` for this particular call but they can be merged. 

What about cases where the maps share keys?

Reply:
In cases where the maps share keys, I would think the most recent key should win. So here, the key passed to the `Info` function would overwrite the key created when the logger was created.